### ISSUE TITLE
include not parsed/recognized after a title or appendix

### ISSUE THREAD

**Issue by luvarqpp** [asciidoctor/asciidoctor-intellij-plugin#319]
Include macro and "ctrl+click" does not work for some instances. I think that it had worked a few releases before current.

It does in my case work only for last include macro (_Release.adoc in screenshot), but I have not found any clue, why. Permissions on files are same, content mostly similar (first line is always level 2, three =, title, e.t.c.):
![asciidocInclude](https://user-images.githubusercontent.com/48614551/62847998-adb8b800-bcd9-11e9-8259-213ee66d92d3.png)

Sample code, but it needs also "parts" directory and files "_Test[ABC].adoc" there:
```
Some last paragraph from doc and than only appendix from includes follow.

[appendix]
= List of rest entities
include::parts/_TestA.adoc[]
include::parts/_TestB.adoc[]
include::parts/_TestC.adoc[]
```

PS: rendering of include macros are not affected. They render correctly.

AsciiDoc Plugin Version: 0.29.4

IDE Name and Version:
IntelliJ IDEA 2019.2 (Community Edition) Build #IC-192.5728.98, built on July 23, 2019 Runtime version: 11.0.3+12-b304.10 amd64 VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o Linux 5.0.0-23-generic GC: ParNew, ConcurrentMarkSweep Memory: 1910M Cores: 8 Registry: git.explicit.commit.renames.prohibit.multiple.calls=false Non-Bundled Plugins: DBN, Lombook Plugin,  com.oliverlockwood.plugins.jenkinsfile,  im.jeanfrancois.ideaplugins.ideafullscreen, manifold.ij,  org.asciidoctor.intellij.asciidoc, com.intellij.lang.jsgraphql,  org.intellij.scala, Docker, training

**Comment by ahus1** [comment]
@luvarqpp - please add a new line after "= List of rest entities" to make it work. 

As the line starts with a single "=" the the current parser treats the two lines following as a title. 

I recommend adding blank lines between includes as keeping them together might leads to unexpected situations: It might glue together two paragraphs. 

Again, I'll have a look into the parser and see how to handle this differently.

**Comment by luvarqpp** [comment]
Is there any chance that "format code" could "autofix" this type of issue? Or can you use parser settings from preview renderer to match parsing behavior of malformed input?

**Sidenote**: Before reporting this "bug", I was trying to use code format, but with no result. I was confident, that this is problem in asciidoc plugin because it renders preview OK and also my maven pipeline renders it ok. I have not recall the fact that code formater had introduced to my documentation many empty lines after titles (thus this I should take as standard formatting).

PS: Adding one empty line after title has indeed fixed my problem.

**Comment by ahus1** [comment]
I've committed a change that support: 
* includes after level 0 headers work
* no longer interprets the lines following a level 0 header when it has a type (like [appendix])

This will be part of an upcoming pre-release.

**Answering the sidenote:**
The plugin aims to implement a JFlex lexer plus a parser that comes close to the current Asciidoctor implementation. It does this IntelliJ-style and aims to be "good enough" to support reformatting and navigation to references. 

Asciidoctor takes a different approach to provide an implementation close to the original Python Asciidoc implementation, and it has to do that to support all of the features AsciiDoc contains. It currently provides an API to receive line numbers on block level, and we use this in some places. 

TL;DR: the Asciidoctor implementation is the "leading" implementation. If you find a problem that reformatting or navigation doesn't work, please create a ticket.

**Comment by ahus1** [comment]
pre-release 0.29.5 is now available.
